examples/informer/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/fields"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	// Only required to authenticate against GKE clusters
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+func main() {
+	kubeconfig := flag.String("kubeconfig", "", "Path to a kube config. Only required if out-of-cluster.")
+	flag.Parse()
+
+	// Create the client config. Use kubeconfig if given, otherwise assume in-cluster.
+	config, err := buildConfig(*kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	stop := make(chan struct{}, 1)
+	source := cache.NewListWatchFromClient(
+		clientset.CoreClient,
+		"pods",
+		api.NamespaceAll,
+		fields.Everything())
+
+	store, controller := cache.NewInformer(
+		source,
+
+		// The object type.
+		&v1.Pod{},
+
+		// resyncPeriod
+		// Every resyncPeriod, all resources in the cache will retrigger events.
+		// Set to 0 to disable the resync.
+		time.Second*0,
+
+		// Your custom resource event handlers.
+		cache.ResourceEventHandlerFuncs{
+			// Takes a single argument of type interface{}.
+			// Called on controller startup and when new resources are created.
+			AddFunc: create,
+
+			// Takes two arguments of type interface{}.
+			// Called on resource update and every resyncPeriod on existing resources.
+			UpdateFunc: update,
+
+			// Takes a single argument of type interface{}.
+			// Called on resource deletion.
+			DeleteFunc: delete,
+		})
+
+	// store can be used to List and Get
+	// NEVER modify objects from the store. It's a read-only, local cache.
+	fmt.Println("listing pods from store:")
+	for _, obj := range store.List() {
+		pod := obj.(*v1.Pod)
+
+		// This will likely be empty the first run, but may not
+		fmt.Printf("%#v\n", pod)
+	}
+
+	// the controller run starts the event processing loop
+	go controller.Run(stop)
+
+	// and now we block on a signal
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	s := <-signals
+	fmt.Printf("received signal %#v, exiting...\n", s)
+	close(stop)
```

Comment:
I would highlight here that the controller will likely not get to actually react to the closed channel and get preempted by the `os.Exit` on the next line. Alternatively include your trick exploiting the fact that `controller.Run` is blocking and passing a `stopped` channel into the controller goroutine that is closed when Run returns.